src/arithmetic/boolean_funcs/boolean_funcs.c

```diff
@@ -0,0 +1,280 @@
+/*
+* Copyright 2018-2019 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include "boolean_funcs.h"
+#include "../func_desc.h"
+#include "../../util/arr.h"
+#include <assert.h>
+
+#define CONTAINS_NULL 2 // Macro used for efficiently evaluating 3-valued truth table
+
+SIValue AR_AND(SIValue *argv, int argc) {
+	// false AND null evaluates to false ; all other null comparisons evaluate to null
+
+	SIValue a = argv[0];
+	SIValue b = argv[1];
+
+	int a_val = SIValue_IsNull(a) ? CONTAINS_NULL : a.longval;
```

Comment:
what if a.longval or b.longval = 2?

Reply:
These functions only accept booleans and NULLs, so any longval other than 0 or 1 is impossible.